float64[] q
---
float64[] T1
float64[] T2
float64[] T3
float64[] T4
float64[] T5
float64[] T6
